VO has no Id

I'm trying to update VO config and I encounter the following error:

> Buildfile: build.xml
> 
> init:
> 
> define.tasks:
> 
> update.vo.config:
>         [echo] Updating VO parameters in /home/pigay/cdb/cfg/grid/*
> [quattor-voconfig] Retrieving configuration for VO aegis
> [quattor-voconfig] Error parsing VO ID cars
> (http://operations-portal.egi.eu/xml/voIDCard/public/all/true)
> 
> BUILD FAILED
> /home/pigay/cdb/quattor.build.xml:435: BUILD FAILED : Invalid
> configuration: VO has no Id

Seems to be a known issue (at least @Pansanel knows it)

I gave another try. It seems to be better, but it ends with the following exception:

```
[quattor-voconfig] VO configuration for VO lofar (ID=251):
[quattor-voconfig]     VOMS Server: null (VOMS port=30019, voms-admin=true)
[quattor-voconfig] Writing template for VO lofar (/home/pigay/cdb/cfg/grid/umd-3/vo/params/lofar.tpl)
[quattor-voconfig]     Suffix 'yzh' not unique for FQAN /surveys (attempt 1)
[quattor-voconfig] VO configuration for VO lsgrid (ID=1390):
[quattor-voconfig]     VOMS Server: null (VOMS port=30018, voms-admin=true)
[quattor-voconfig] Writing template for VO lsgrid (/home/pigay/cdb/cfg/grid/umd-3/vo/params/lsgrid.tpl)
[quattor-voconfig]     INFO: VO lsgrid has no specific FQAN defined
[quattor-voconfig] VO configuration for VO lsst (ID=10281):
[quattor-voconfig]     VOMS Server: null (VOMS port=15003, voms-admin=true)
[quattor-voconfig]     VOMS Server: null (VOMS port=15003, voms-admin=false)
[quattor-voconfig]     VOMS Server: null (VOMS port=15003, voms-admin=true)
[quattor-voconfig] Writing template for VO lsst (/home/pigay/cdb/cfg/grid/umd-3/vo/params/lsst.tpl)
[quattor-voconfig]     INFO: VO lsst has no specific FQAN defined
[quattor-voconfig] VO configuration for VO lz (ID=10320):
[quattor-voconfig]     VOMS Server: null (VOMS port=15002, voms-admin=true)
[quattor-voconfig] Writing template for VO lz (/home/pigay/cdb/cfg/grid/umd-3/vo/params/lz.tpl)

BUILD FAILED
java.lang.StringIndexOutOfBoundsException: String index out of range: 3
    at java.lang.String.substring(String.java:1946)
    at org.quattor.ant.VOConfigTask$VOConfig.setAccountPrefix(VOConfigTask.java:641)
    at org.quattor.ant.VOConfigTask$VOConfig.getAccountPrefix(VOConfigTask.java:601)
    at org.quattor.ant.VOConfigTask$VOConfig.writeVOTemplate(VOConfigTask.java:700)
    at org.quattor.ant.VOConfigTask$VOConfig.access$000(VOConfigTask.java:562)
    at org.quattor.ant.VOConfigTask.execute(VOConfigTask.java:277)
    at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:288)
    at sun.reflect.GeneratedMethodAccessor1.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:622)
    at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
    at org.apache.tools.ant.Task.perform(Task.java:348)
    at org.apache.tools.ant.Target.execute(Target.java:357)
    at org.apache.tools.ant.Target.performTasks(Target.java:385)
    at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1337)
    at org.apache.tools.ant.Project.executeTarget(Project.java:1306)
    at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
    at org.apache.tools.ant.Project.executeTargets(Project.java:1189)
    at org.apache.tools.ant.Main.runBuild(Main.java:758)
    at org.apache.tools.ant.Main.startAnt(Main.java:217)
    at org.apache.tools.ant.launch.Launcher.run(Launcher.java:257)
    at org.apache.tools.ant.launch.Launcher.main(Launcher.java:104)

Total time: 2 seconds
```

Maybe it's not related to the preceding?

--- a/src/main/java/VOConfigTask.java
+++ b/src/main/java/VOConfigTask.java
@@ -71,6 +71,9 @@ public class VOConfigTask extends Task {
     /* URI for VO ID card source */
     protected String voIdCardsUri = null;
 
+    /* Name of the attribute containing the VO ID */
+    protected String voIdAttributeName = "CIC_ID";
+
     /* Name of the template containing the list of all defined VOs */
     protected String allVosTemplate = "allvos";
 
@@ -143,6 +146,21 @@ public class VOConfigTask extends Task {
      */
     public void setvoIdCardsUri(String voIdCardsUri) {
         this.voIdCardsUri = voIdCardsUri;
+    }
+
+    /**
+     * Set the name of the attribute containing the VO ID
+     *
+     * @param voIdAttributeName
+     *            String containing the name of the attribute defining
+     *            the VO ID
+     *
+     */
+    public void setvoIdAttributeName(String voIdAttributeName) {
+        this.voIdAttributeName = voIdAttributeName;
+        if ( this.voIdAttributeName.equals("CIC_ID") ) {
+            System.err.println("    WARNING: The use of the CIC_ID attribute is deprecated. The attribute named Serial should be used instead.");
+        }
     }
 
     /**
@@ -261,7 +279,9 @@ public class VOConfigTask extends Task {
             URL voIdCardsUrl = new URL(voIdCardsUri);
             InputStream urlstream = voIdCardsUrl.openStream();
             SAXParser parser = factory.newSAXParser();
-            parser.parse(urlstream, new VOCardHandler());        
+            VOCardHandler vocardhandler = new VOCardHandler();
+            vocardhandler.setVoIdAttributeName(this.voIdAttributeName);
+            parser.parse(urlstream, vocardhandler);
         } catch (MalformedURLException e) { 
             System.out.println("Invalid format used for specifying the source of VO ID cards (voIdCardsUri): "+voIdCardsUri);
             throw new BuildException("BUILD FAILED : " + e.getMessage());
@@ -384,6 +404,7 @@ public class VOConfigTask extends Task {
         protected VOMSEndpoint vomsEndpoint = null;
         protected VOMSFqan fqan = null;
         protected String data = null;
+        protected String voIdAttributeName = null;
 
         /**
          * Start of new element
@@ -399,13 +420,19 @@ public class VOConfigTask extends Task {
                 voName = voName.toLowerCase();
                 System.out.println("Retrieving configuration for VO "+voName);
                 voConfig = new VOConfig();
-                String voId = attributes.getValue("CIC_ID");
+                if ( voIdAttributeName == null) {
+                    throw new SAXException("Invalid configuration: the voIdAttributeName is not set");
+                }
+                String voId = attributes.getValue(voIdAttributeName);
                 if ( voId == null ) {
                     throw new SAXException("Invalid configuration: VO has no Id");
                 }
                 voConfig.setName(voName);
-                voConfig.setId(Integer.parseInt(voId));
-                
+                if ( this.voIdAttributeName.equals("CIC_ID") ) {
+                    voConfig.setId(Integer.parseInt(voId));
+                } else {
+                    voConfig.setId(Integer.parseInt(voId)*10+40);
+                }
             } else if ( qName.equals("VOMSServers") ) {
                 sectionVOMSServers = true;
                 
@@ -565,8 +592,20 @@ public class VOConfigTask extends Task {
             }
         }
 
+       /**
+        * Set the name of the attribute containing the VO ID
+        *
+        * @param voIdAttributeName
+        *            String containing the name of the attribute defining
+        *            the VO ID
+        *
+        */
+
+        public void setVoIdAttributeName(String voIdAttributeName) {
+            this.voIdAttributeName = voIdAttributeName;
+        }
     }
-        
+
 
     // Class representing a VO
 

